src/Arcus.Scripting.IntegrationAccount/Scripts/Set-AzIntegrationAccountAgreements.ps1

```diff
@@ -0,0 +1,87 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $ResourceGroupName = $(throw "Resource group name is required"),
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the Integration Account is required"),
+    [parameter(Mandatory = $false)][string] $AgreementFilePath = $(if ($AgreementsFolder -eq '') { throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'" }),
+    [parameter(Mandatory = $false)][string] $AgreementsFolder = $(if ($AgreementFilePath -eq '') { throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'" }),
+    [Parameter(Mandatory = $false)][string] $ArtifactsPrefix = ''
+)
+
+if ($AgreementFilePath -ne '' -and $AgreementsFolder -ne '') {
+    throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'"
+}
+
+function UploadAgreement {
+    param(
+        [Parameter(Mandatory = $true)][System.IO.FileInfo] $Agreement
+    )
+
+    $agreementData = Get-Content -Raw -Path $Agreement.FullName | ConvertFrom-Json
```

Comment:
I'm not really sure what kind of depth to use here, since agreements can be quite complex and have deep structures. I don't think it will overload, according to the docs the default is 1024: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json?view=powershell-7.1

Also did a quick search through all scripts and can't find the depth being used anywhere else. Doesn't have to mean anything of course, but the overall implementations seem to not use the depth anywhere.